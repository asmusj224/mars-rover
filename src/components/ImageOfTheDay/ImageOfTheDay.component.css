.image-of-the-day {
  margin-top: 2rem;
  margin-bottom: 2rem; }
  .image-of-the-day__image {
    display: block;
    max-width: 100%;
    height: 35rem;
    margin: auto;
    border-radius: 3px; }
  .image-of-the-day__rover {
    display: flex;
    align-items: center;
    justify-content: center;
    color: #a9a9a9;
    font-size: 14px; }
