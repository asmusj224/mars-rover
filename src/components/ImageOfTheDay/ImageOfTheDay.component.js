import React from 'react';
import './ImageOfTheDay.component.css';
const ImageOfTheDay = ({ roverImage, roverName, roverCameraName }) => {
    return (
        <div className="image-of-the-day">
            <img className="image-of-the-day__image" src={roverImage} alt={roverName} />
            <p className="image-of-the-day__rover">{roverName} - {roverCameraName}</p>
        </div>
    );
}

export default ImageOfTheDay;