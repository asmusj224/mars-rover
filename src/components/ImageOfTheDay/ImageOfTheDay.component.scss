@import 'styles/_colors.scss';
.image-of-the-day {
    margin-top: 2rem;
    margin-bottom: 2rem;
    &__image {
        display: block;
        max-width: 100%;
        height: 35rem;
        margin: auto;
        border-radius: 3px;
    }
    &__rover {
        display: flex;
        align-items: center;
        justify-content: center;
        color: $darkgray;
        font-size: 14px;
    }
}