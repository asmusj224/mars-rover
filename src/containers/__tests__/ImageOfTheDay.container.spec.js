import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'

import ImageOfTheDay from '../ImageOfTheDay.container';

const middlewares = [];
const mockStore = configureMockStore(middlewares);
const initialState = {
    imageOfTheDay: {
        roverImage: '',
        roverName: '',
        roverCameraName: ''
    }
};
const store = mockStore(initialState);


describe('<ImageOfTheDay />', () => {
    it('renders correctly', () => {
        const component = renderer.create(
            <Provider store={store}>
                <ImageOfTheDay />
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});