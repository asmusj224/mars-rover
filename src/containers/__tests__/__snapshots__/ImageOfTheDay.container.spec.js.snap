// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageOfTheDay /> renders correctly 1`] = `
<div
  className="image-of-the-day"
>
  <img
    alt=""
    className="image-of-the-day__image"
    src=""
  />
  <p
    className="image-of-the-day__rover"
  >
    
     - 
    
  </p>
</div>
`;
